common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -385,14 +389,27 @@ private static CommandResult executeCommand(@SuppressWarnings(WarningType.rawtyp
                         command.getParameters().getCorrelationId());
             }
         } else /* baseException == null */ {
-            if (result != null && result instanceof AcquireTokenResult) {
+            if (result instanceof AcquireTokenResult) {
                 //Handler handler, final BaseCommand command, BaseException baseException, AcquireTokenResult result
                 commandResult = getCommandResultFromTokenResult(baseException, (AcquireTokenResult) result,
                         command.getParameters().getCorrelationId());
             } else {
+
                 //For commands that don't return an AcquireTokenResult
                 commandResult = new CommandResult(CommandResult.ResultStatus.COMPLETED, result,
                         command.getParameters().getCorrelationId());
+
+                //If any record needs a refresh, those records need to be refreshed.
+                if (result instanceof List
```

Comment:
This class seems to contain a bit more generic command implementation and handling logic. Is there any place that is more specific to silent token requests that would be more appropriate for this logic?

Reply:
+1...I think all the logic can go either in the controllers or the specific command class for silent token requests called as `SilentTokenCommand`

Reply:
Outdated